dissect/extfs/extfs.py

```diff
@@ -117,25 +116,22 @@ def journal(self) -> JDB2:
 
         return self._journal
 
-    def get(self, path: str | int, node: INode | None = None) -> INode:
-        if isinstance(path, int):
-            return self.get_inode(path)
+    def get(self, path_or_inum: str | int, node: INode | None = None) -> INode:
+        if isinstance(path_or_inum, int):
+            return self.get_inode(path_or_inum)
 
         node = node if node else self.root
-        parts = path.split("/")
-        for part_num, part in enumerate(parts):
+        parts = path_or_inum.split("/")
+        for _, part in enumerate(parts):
             if not part:
                 continue
 
-            while node.filetype == stat.S_IFLNK and part_num < len(parts):
```

Comment:
Should we not try to resolve symlinks within `.get()` if we can?

Reply:
I implemented the resolve symlinks inside `::get()`. While it is only a couple of lines, it muddles the implementation and feels out of place, and the boolean toggle also feels awkward. The idea that this has to be implemented for more filesystems is also not very appealing.

I think I have an idea which provides interactive symlink resolve without a target  for multiple file systems. Will defer that to the XFS ticket (will create one).

Reply:
I share the same opinion as JSCU-CNI